DependencyInjection/ClientFactory.php

```diff
@@ -389,6 +389,12 @@ protected function setupUserDetection(Client $client, TokenStorageInterface $tok
             $user = $token->getUser();
 
             if ($user instanceof UserInterface) {
+                // Symfony 5.3 introduced 'getUserIdentifier' to replace 'getUsername'
+                // Symfony 6.0 removed 'getUsername'
+                if (method_exists(UserInterface::class, 'getUserIdentifier')) {
+                    return ['id' => $user->getUserIdentifier()];
+                }
+
                 return ['id' => $user->getUsername()];
```

Comment:
Perhaps this should also be wrapped in a `method_exists` check (in the case that the name changes again in the future and the `getUserIdentifier` clause falls through)?